lib/Db/MessageSnoozeMapper.php

```diff
@@ -39,6 +39,34 @@ public function __construct(IDBConnection $db, private ITimeFactory $time) {
 		parent::__construct($db, 'mail_messages_snoozed', MessageSnooze::class);
 	}
 
+	/**
+	 * Returns srcMailboxId (before snooze) for message
+	 * Return null if no entry for message or srcMailboxId is null
+	 *
+	 * @param string $messageId
+	 *
+	 * @return int|null
+	 */
+	public function getSrcMailboxId(string $messageId): ?int {
+		$qb = $this->db->getQueryBuilder();
+
+		$select = $qb->select('src_mailbox_id')
+			->from($this->getTableName())
+			->where($qb->expr()->eq(
+				'message_id',
+				$qb->createNamedParameter($messageId, IQueryBuilder::PARAM_STR),
+				IQueryBuilder::PARAM_STR,
+			));
+
+		$result = $select->executeQuery();
+		$rows = $result->fetchAll();
```

Comment:
```suggestion
		$rows = $result->fetch();
```
You can use fetch if you expect only one row.